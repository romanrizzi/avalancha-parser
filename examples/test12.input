
-- Funciones (signatura sin reglas)

fun foo1 -- sin signatura ni reglas (tiene aridad 0)

-- con signatura sin reglas, aridad 0
fun foo2 : -> _
fun foo3 : -> xx

-- con signatura sin reglas, aridad 1
fun foo4 : _ -> _
fun foo5 : x -> _
fun foo6 : _ -> y
fun foo7 : x -> y

-- con signatura sin reglas, aridad 2
fun foo8:_,_->_
fun foo9:x,_->_
fun foo10:_,y->_
fun foo11:x,y->_
fun foo12:_,_->z
fun foo13:x,_->z
fun foo14:_,y->z
fun foo15:x,y->z

