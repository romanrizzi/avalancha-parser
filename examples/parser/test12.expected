[
   "program",
   [
      [
         "fun",
         "foo1",
         [
            "sig",
            [],
            "_"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo2",
         [
            "sig",
            [],
            "_"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo3",
         [
            "sig",
            [],
            "xx"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo4",
         [
            "sig",
            [
               "_"
            ],
            "_"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo5",
         [
            "sig",
            [
               "x"
            ],
            "_"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo6",
         [
            "sig",
            [
               "_"
            ],
            "y"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo7",
         [
            "sig",
            [
               "x"
            ],
            "y"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo8",
         [
            "sig",
            [
               "_",
               "_"
            ],
            "_"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo9",
         [
            "sig",
            [
               "x",
               "_"
            ],
            "_"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo10",
         [
            "sig",
            [
               "_",
               "y"
            ],
            "_"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo11",
         [
            "sig",
            [
               "x",
               "y"
            ],
            "_"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo12",
         [
            "sig",
            [
               "_",
               "_"
            ],
            "z"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo13",
         [
            "sig",
            [
               "x",
               "_"
            ],
            "z"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo14",
         [
            "sig",
            [
               "_",
               "y"
            ],
            "z"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "foo15",
         [
            "sig",
            [
               "x",
               "y"
            ],
            "z"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ]
   ],
   []
]
